bpfd/src/oci_utils/image_manager.rs

```diff
@@ -471,4 +483,20 @@ mod tests {
 
         assert!(matches!(result, Err(ImageError::ByteCodeImageNotfound(_))));
     }
+
+    #[rstest(input, output,
+        case("busybox", PathBuf::from("/var/lib/bpfd/io.bpfd.image.content/docker.io/library/busybox/latest")),
+        case("quay.io/busybox", PathBuf::from("/var/lib/bpfd/io.bpfd.image.content/quay.io/busybox/latest")),
+        case("docker.io/test:tag", PathBuf::from("/var/lib/bpfd/io.bpfd.image.content/docker.io/library/test/tag")),
+        case("quay.io/test:5000", PathBuf::from("/var/lib/bpfd/io.bpfd.image.content/quay.io/test/5000")),
+        case("test.com/repo:tag", PathBuf::from("/var/lib/bpfd/io.bpfd.image.content/test.com/repo/tag")),
+        case("test.com/repo@sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+        PathBuf::from("/var/lib/bpfd/io.bpfd.image.content/test.com/repo/sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff")),
+    )]
+    fn test_good_image_content_path(input: &str, output: PathBuf) {
+        let good_reference: Reference = input.parse().unwrap();
+
+        let image_content_path = get_image_content_dir(good_reference);
+        assert_eq!(image_content_path, output);
+    }
```

Comment:
```suggestion
    #[test]
    fn test_good_image_content_path() {
        struct Case {
            input: &'static str,
            output: PathBuf,
        }

        let tt = vec![
            Case{input: "busybox", output: PathBuf::from("/var/lib/bpfd/io.bpfd.image.content/docker.io/library/busybox/latest")},
            Case{input:"quay.io/busybox", output: PathBuf::from("/var/lib/bpfd/io.bpfd.image.content/quay.io/busybox/latest")},
            Case{input:"docker.io/test:tag", output: PathBuf::from("/var/lib/bpfd/io.bpfd.image.content/docker.io/library/test/tag")},
            Case{input:"quay.io/test:5000", output: PathBuf::from("/var/lib/bpfd/io.bpfd.image.content/quay.io/test/5000")},
            Case{input:"test.com/repo:tag", output: PathBuf::from("/var/lib/bpfd/io.bpfd.image.content/test.com/repo/tag")},
            Case{
                input:"test.com/repo@sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
                output: PathBuf::from("/var/lib/bpfd/io.bpfd.image.content/test.com/repo/sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"),
            }
        ];

        for t in tt {
            let good_reference: Reference = t.input.parse().unwrap();

            let image_content_path = get_image_content_dir(good_reference);
            assert_eq!(image_content_path, t.output);
        }
    }
```

Achieves the same thing without depending on another crate

Reply:
Sounds good to me!